engine/src/main/java/com/datatorrent/stram/client/StramClientUtils.java

```diff
@@ -519,28 +520,77 @@ public static FileSystem newFileSystemInstance(Configuration conf) throws IOExce
     }
   }
 
+  /**
+   * Helper function used by both getApexDFSRootDir and getDTDFSRootDir to process dfsRootDir as per the required logic
+   *
+   * @param fs
+   * @param conf
+   * @param dfsRootDir
+   * @param useImpersonated
+   * @param processRelativePath
+   * @return
+   */
+  private static Path processDFSRootDirHelper(FileSystem fs, Configuration conf, String dfsRootDir, boolean useImpersonated,
+      boolean processRelativePath)
+  {
+    try {
+      String userShortName = useImpersonated ?
+          UserGroupInformation.getCurrentUser().getShortUserName() : UserGroupInformation.getLoginUser().getShortUserName();
+      boolean userNameReplaced = false;
+      if (dfsRootDir.contains(DT_DFS_USER_NAME)) {
+        dfsRootDir = dfsRootDir.replace(DT_DFS_USER_NAME, userShortName);
+        conf.set(DT_DFS_ROOT_DIR, dfsRootDir);
+        userNameReplaced = true;
```

Comment:
Hmmm, so you want the userShortName to be used twice under certain circumstances? As an example if the value of apex.app.dfsRootDir is "%USER_NAME%" (the actual string with % chars) and say the current user userShortName is peter then you will get /user/peter/peter as the dfsRootDir to be used by the current app. If that's okay I will remove the check for userNameReplaced